src/LndClient/RPC/Katip.hs

```diff
@@ -127,3 +130,29 @@ ensureHodlInvoice env req =
               AddInvoice.paymentRequest = Invoice.paymentRequest x,
               AddInvoice.addIndex = Invoice.addIndex x
             }
+
+closeChannelSync ::
+  (KatipContext m, MonadUnliftIO m) =>
+  LndEnv ->
+  CloseChannelRequest ->
+  m (Either LndError ())
+closeChannelSync = closeChannelRecursive 10
+  where
+    closeChannelRecursive (0 :: Int) env _ = do
+      $(logTM)
+        (newSeverity env ErrorS Nothing Nothing)
+        "Channel couldn't be closed."
+      return $ Left $ LndError "Cannot close channel"
+    closeChannelRecursive n env req = do
+      mVar <- newEmptyMVar
+      _ <-
+        Util.spawnLink $
+          closeChannel
```

Comment:
Please check that channel is open (in `listChannels`) before spawning this. If channel is not in list - it does not make sense to spawn close procedure

Reply:
All changes are the same for both (Katip and Silent)

Reply:
Agree, this was my initial plan.
Done